qldbdriver/result.go

```diff
@@ -66,9 +68,51 @@ func (result *Result) getNextPage() error {
 	result.pageValues = nextPage.Page.Values
 	result.pageToken = nextPage.Page.NextPageToken
 	result.index = 0
+	result.updateMetrics(nextPage)
 	return nil
 }
 
+func (result *Result) updateMetrics(fetchPageResult *qldbsession.FetchPageResult) {
+	if fetchPageResult.ConsumedIOs != nil {
+		if result.consumedIOs == nil {
+			result.consumedIOs = &IOUsage{new(int64), new(int64)}
+		}
+		*result.consumedIOs.readIOs += *fetchPageResult.ConsumedIOs.ReadIOs
+		*result.consumedIOs.writeIOs += *fetchPageResult.ConsumedIOs.WriteIOs
+	}
+
+	if fetchPageResult.TimingInformation != nil {
+		if result.timingInformation == nil {
+			result.timingInformation = &TimingInformation{new(int64)}
+		}
+		*result.timingInformation.processingTimeMilliseconds += *fetchPageResult.TimingInformation.ProcessingTimeMilliseconds
+	}
+}
+
+// GetConsumedIOs returns the statement statistics for the current number of read IO requests that were consumed. The statistics are stateful.
+func (result *Result) GetConsumedIOs() *IOUsage {
+	if result.consumedIOs == nil {
```

Comment:
And so for example, you could remove this check too :)

Reply:
Removed this check.